pages/blog/release-notes-2.5.0.md

```diff
@@ -0,0 +1,94 @@
+---
+title: AsyncAPI Spec 2.5.0 Release Notes
+date: 2022-09-20T16:41:00+01:00
+type: Communication
+tags:
+  - Specification
+  - Release Notes
+cover: /img/posts/release-notes-2.5.0/cover.webp
+authors:
+  - name: Vladimír Gorej
+    photo: /img/avatars/char0n.webp
+    link: https://twitter.com/vladimirgorej
+    byline: Open Source Software Engineer, OSS contributor, author and content creator.
+excerpt: "AsyncAPI 2.5 is now released. This brings really helpful additions, such as extended Server Variables reusability, and many clarifications"
+featured: true
+---
+
+The new version of the AsyncAPI specification - 2.5.0 - is now available.
+
+> This is a minor release, and it doesn't bring any breaking changes. You can switch to it by modifying the following value in your AsyncAPI file `asyncapi: '2.4.0'` into `asyncapi: '2.5.0'`
+
+## Server Variables can be now referenced from inside the Server Object 
+
+Previous AsyncAPI 2.4.0 release came with support of [defining Server Variable Objects as reusable components](https://www.asyncapi.com/blog/release-notes-2.4.0#server-variables-can-be-now-referenced-from-components).
+Unfortunately AsyncAPI 2.4.0 doesn't allow to reference reusable server variable objects from within the `Server Object`.
+So we ended up in a situation that reusable server variable objects can be defined, but they actually cannot be reused.
+AsyncAPI 2.5.0 remedies this deficiency by allowing to use `Reference Object` whenever `Server Variable Object` can be used inside `Server.variables` field.
+
+Following example is now fully compliant with the specification:
+
+```yaml
+asyncapi: 2.4.0
+servers:
+  development:
+    $ref: '#/components/servers/myserver'
+  production:
+    $ref: '#/components/servers/myserver'
+components:
+  servers:
+    myserver:
+      url: "{stage}.my-server.com:{port}"
+      protocol: ws
+      variables:
+        stage:
+          $ref: "#/components/serverVariables/stage"
+        port:
+          $ref: "#/components/serverVariables/port"
+  serverVariables:
+    stage:
+      default: dev
+    port:
+      enum: [5000, 6000]
+      default: 5000
+```
+
+This new feature was contributed by [Vladimír Gorej](https://github.com/char0n). For more detail, see [Vladimír's `/spec #776` pull request](https://github.com/asyncapi/spec/pull/776) and the [GitHub issue where Vladimír's change to `serverVariables ` was discussed](https://github.com/asyncapi/spec/issues/775).
+
+## Decorating Server Object with tags
+
+placeholder for description
```

Comment:
@smoya can I ask you to provide section describing your contribution? Thanks!

Reply:
Here we go:

Tags are used to categorize or group resources. 2.5.0 introduces support for defining tags at Server level, which can be used for many use cases like grouping servers by their features: environment name, cluster name, visibility, owner, etc.

For example:

```yaml
asyncapi: '2.5.0'
info:
  title: Streetlights Kafka API
  version: '1.0.0'
servers:
  scram-connections:
    url: test.mykafkacluster.org:18092
    protocol: kafka-secure
    description: Test broker secured with scramSha256
    security:
      - saslScram: []
    tags:
      - name: "env:test-scram"
        description: "This environment is meant for running internal tests through scramSha256"
      - name: "kind:remote"
        description: "This server is a remote server. Not exposed by the application"
      - name: "visibility:private"
        description: "This resource is private and only available to certain users"  
  mtls-connections:
    url: test.mykafkacluster.org:28092
    protocol: kafka-secure
    description: Test broker secured with X509
    security:
      - certs: []
    tags:
      - name: "env:test-mtls"
        description: "This environment is meant for running internal tests through mtls"
      - name: "kind:remote"
        description: "This server is a remote server. Not exposed by the application"
      - name: "visibility:private"
        description: "This resource is private and only available to certain users"
channels: {}
```

This new feature was contributed by [Sergio Moya](https://github.com/smoya). For more detail, see [Sergio's /spec #809 pull request](https://github.com/asyncapi/spec/pull/809) and the [Github issue where Sergio's change to was discussed](https://github.com/asyncapi/spec/issues/654).

Reply:
Incorporated into draft blog post. Thank you!